swap/swap.go

```diff
@@ -155,22 +153,90 @@ func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 	log.Debug(fmt.Sprintf("balance for peer %s after accounting: %s", peer.ID().String(), strconv.FormatInt(peerBalance, 10)))
 
 	//check if balance with peer is over the payment threshold
-	if peerBalance >= s.paymentThreshold {
-		//if so, send cheque request message
-		log.Warn(fmt.Sprintf("balance for peer %s went over the payment threshold %v, requesting cheque", peer.ID().String(), s.paymentThreshold))
-		chequeRequestMessage := &ChequeRequestMsg{
-			Beneficiary: s.owner.address,
-		}
-		err = swapPeer.Send(context.TODO(), chequeRequestMessage)
-
+	if peerBalance <= -1*s.paymentThreshold {
+		//if so, send cheque
+		log.Warn(fmt.Sprintf("balance for peer %s went over the payment threshold %v, sending cheque", peer.ID().String(), s.paymentThreshold))
+		err = s.sendCheque(peer.ID())
 		if err != nil {
-			log.Error(fmt.Sprintf("error while sending cheque request message to peer %s: %s", peer.ID().String(), err.Error()))
+			log.Error(fmt.Sprintf("error while sending cheque to peer %s: %s", peer.ID().String(), err.Error()))
 		}
 	}
 
 	return
 }
 
+func (s *Swap) sendCheque(peer enode.ID) error {
```

Comment:
This function should be behind a semaphore lock.

Reply:
Done.